server/websockets.js

```diff
@@ -46,6 +47,11 @@ class WebsocketServer {
 
     this.nydus.on('connection', socket => {
       this.nydus.subscribeClient(socket, '/status', { users: this.connectedUsers })
+
+      // TODO(2Pac): Only do this for Electron clients
+      if (matchmakingStatusInstance) {
```

Comment:
The middleware for this exports a function to check this, so you should be able to do this in this PR. The request that started the connection is stored as `socket.conn.request`

Reply:
Yes, but that middleware checks for `X-Shield-Battery-Client` header from what I can tell. And that header is not being set correctly for a while now on websocket connections ^^.

It's a whole thing, but basically we need to upgrade `nydus` to make it work. Either by waiting for `engine.io` to release a new version with my fix in (which has not been merged in yet), or by using the `attach()` method to construct a new server instance instead of using the constructor directly (which doesn't pass options correctly).

I _think_ I got this, but maybe I'm missing something?